src/components/text-input/TextInput.js

```diff
@@ -46,29 +45,28 @@ const TextInput = ({
     labelTooltip,
     ...rest
 }: Props) => {
-    const classes = classNames(className, 'text-input-container', {
-        'show-error': !!error,
+    const classes = classNames(className, 'bdl-TextInput-container', {
+        'bdl-hasError': !!error,
     });
 
     return (
         <div className={classes}>
             <Label
+                className="bdl-TextInput-label"
                 hideLabel={hideLabel}
                 showOptionalText={!hideOptionalLabel && !isRequired}
                 text={label}
                 tooltip={labelTooltip}
             >
-                {!!description && <i className="text-input-description">{description}</i>}
+                {!!description && <i className="bdl-TextInput-description">{description}</i>}
                 <Tooltip isShown={!!error} position={errorPosition || 'middle-right'} text={error || ''} theme="error">
-                    <input ref={inputRef} required={isRequired} {...rest} />
+                    <input {...rest} className="bdl-TextInput-input" ref={inputRef} required={isRequired} />
```

Comment:
Moving `{...rest}` seems like a risky-ish change. Moving it doesn't seem related to styling, since the class names are stripped out in the method signature.